File: tests/goldens/java/valid_billing_policy.java
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.DisplayName;
import static org.junit.jupiter.api.Assertions.assertEquals;

import com.example.BillingPolicy;

public class BillingpolicyTest {

    // Source: tests/fixtures/valid_billing_policy.tdd:6
    @Test
    @DisplayName("grandfathers loyal customers onto the pro cap")
    public void testGrandfathersLoyalCustomersOntoTheProCap() {
        // Input: {'account': {'plan': 'legacy', 'yearsActive': 7}, 'usage'...
        var input = java.util.Map.of("account", java.util.Map.of("plan", "legacy", "yearsActive", 7), "usage", java.util.Map.of("projects", 18, "seats", 4));
        var result = BillingPolicy.quoteSubscription(input);
        // Expected: {'tier': 'pro', 'monthlyUsd': 49, 'requiresReview': False}
        assertEquals("pro", result.getTier());
        assertEquals(49, result.getMonthlyusd());
        assertEquals(false, result.getRequiresreview());
    }

    // Source: tests/fixtures/valid_billing_policy.tdd:20
    @Test
    @DisplayName("flags enterprise usage before charging")
    public void testFlagsEnterpriseUsageBeforeCharging() {
        // Input: {'account': {'plan': 'team', 'yearsActive': 1}, 'usage': ...
        var input = java.util.Map.of("account", java.util.Map.of("plan", "team", "yearsActive", 1), "usage", java.util.Map.of("projects", 91, "seats", 42));
        var result = BillingPolicy.quoteSubscription(input);
        // Expected: {'tier': 'enterprise', 'monthlyUsd': None, 'requiresRevie...
        assertEquals("enterprise", result.getTier());
        assertEquals(null, result.getMonthlyusd());
        assertEquals(true, result.getRequiresreview());
        assertEquals("seat_count", result.getReason());
    }

}

File: tests/goldens/kotlin/valid_billing_policy.kt
package com.example

import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

class BillingpolicyTest {

    // Source: tests/fixtures/valid_billing_policy.tdd:8
    @Test
    fun `grandfathers loyal customers onto the pro cap`() {
        // Input: {'account': {'plan': 'legacy', 'yearsActive': 7}, 'usage'...
        val input = mapOf("account" to mapOf("plan" to "legacy", "yearsActive" to 7), "usage" to mapOf("projects" to 18, "seats" to 4))
        val result = BillingPolicy.quoteSubscription(input)
        // Expected: {'tier': 'pro', 'monthlyUsd': 49, 'requiresReview': False}
        assertEquals("pro", result.tier)
        assertEquals(49, result.monthlyUsd)
        assertEquals(false, result.requiresReview)
    }

    // Source: tests/fixtures/valid_billing_policy.tdd:22
    @Test
    fun `flags enterprise usage before charging`() {
        // Input: {'account': {'plan': 'team', 'yearsActive': 1}, 'usage': ...
        val input = mapOf("account" to mapOf("plan" to "team", "yearsActive" to 1), "usage" to mapOf("projects" to 91, "seats" to 42))
        val result = BillingPolicy.quoteSubscription(input)
        // Expected: {'tier': 'enterprise', 'monthlyUsd': None, 'requiresRevie...
        assertEquals("enterprise", result.tier)
        assertEquals(null, result.monthlyUsd)
        assertEquals(true, result.requiresReview)
        assertEquals("seat_count", result.reason)
    }

}